fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/lock/InMemoryResourceLockManager.java

```diff
@@ -60,11 +72,29 @@ public void acquire(final String txId, final FedoraId resourceId) {
                         String.format("Cannot update %s because it is being updated by another transaction.",
                                 resourceIdStr));
             }
+            final String estimateParentPath = resourceIdStr.substring(0, resourceIdStr.lastIndexOf('/'));
+            final var actualParent = containmentIndex.getContainerIdByPath(tx, resourceId, false);
+
+            if (!estimateParentPath.equalsIgnoreCase(actualParent.getResourceId())) {
+                // If the expected parent does not match the actual parent, then we have ghost nodes.
+                // Add them as well.
+                LOG.debug("Getting lock for ghost parents as well.");
+                final List<String> ghostPaths = Arrays.stream(estimateParentPath
+                        .replace(actualParent.getResourceId(), "")
+                        .split("/")).filter(a -> !a.isBlank()).collect(Collectors.toList());
+                FedoraId tempParent = actualParent;
+                for (final String part : ghostPaths) {
+                    tempParent = tempParent.resolve(part);
+                    lockedResources.add(tempParent.getResourceId());
```

Comment:
I don't think locking ghost nodes here is safe. The problem is that this code is being executed within a synchronized block for the _child_ resource, which means that there's nothing preventing another transaction from concurrently locking the same resources.